packages/nvidia_nat_mcp/src/nat/plugins/mcp/auth/service_account/provider.py

```diff
@@ -0,0 +1,101 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from pydantic import SecretStr
+
+from nat.authentication.interfaces import AuthProviderBase
+from nat.data_models.authentication import AuthResult
+from nat.data_models.authentication import Credential
+from nat.data_models.authentication import HeaderCred
+from nat.plugins.mcp.auth.service_account.provider_config import MCPServiceAccountProviderConfig
+from nat.plugins.mcp.auth.service_account.token_client import ServiceAccountTokenClient
+
+logger = logging.getLogger(__name__)
+
+
+class MCPServiceAccountProvider(AuthProviderBase[MCPServiceAccountProviderConfig]):
+    """
+    MCP service account authentication provider using OAuth2 client credentials.
+
+    Provides headless authentication for MCP clients using service account credentials.
+    Supports both standard Bearer tokens and custom token prefix formats.
+
+    Example Configuration:
+        authentication:
+          my_service:
+            _type: mcp_service_account
+            client_id: ${SERVICE_ACCOUNT_CLIENT_ID}
+            client_secret: ${SERVICE_ACCOUNT_CLIENT_SECRET}
+            token_url: https://auth.example.com/oauth/token
+            scopes:
+              - api.read
+              - api.write
+            token_prefix: service_account  # Optional
+            service_token: ${SERVICE_TOKEN}  # Optional
```

Comment:
Docs build is failing because of this indentation. Should explicitly be a code block with YAML syntax.